config/opal_config_pthreads.m4

```diff
@@ -251,9 +252,10 @@ AC_PROVIDE_IFELSE([AC_PROG_CC],
                   [OPAL_INTL_POSIX_THREADS_PLAIN_C],
                   [opal_pthread_c_success=1])
 
-AC_PROVIDE_IFELSE([AC_PROG_CXX],
-                  [OPAL_INTL_POSIX_THREADS_PLAIN_CXX],
-                  [opal_pthread_cxx_success=1])
+AS_IF([test "$WANT_MPI_CXX_SUPPORT" = 1],
+      [AC_PROVIDE_IFELSE([AC_PROG_CXX],
```

Comment:
I actually can't find any documentation on `AC_PROVIDE_IFELSE` -- I see that we use it all throughout `opal_config_pthreads.m4`, but I can't quite figure out what it is *for*.  My best guess is that it is "if `AC_PROG_CXX` has been substituted earlier, then substitute $1, otherwise substitute $2."

But if that's correct, it's been used entirely wrong in this m4 file (and likely should be removed).

Specifically: I'm trying to figure out why your new test on `$WANT_MPI_CXX_SUPPORT` is necessary.  And if it is, is the `AC_PROVIDE_IFELSE` unnecessary (or incorrect)?